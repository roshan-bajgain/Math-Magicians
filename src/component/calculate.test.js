// Test for component/calculate.js
import React from 'react';
import { render, screen, act } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Calculator from './calculator';

it('renders correctly', async () => {
  const arr = ['Ac','+/-','%','7','8','9','×','6','5','4','3','2','1','0','÷','−','+','.','='];
  const { asFragment } = render(<Calculator />);
  const buttons = screen.getAllByRole('button');
   act(()=> {
    buttons.forEach((btn)=> {
      userEvent.click(btn);
      expect(arr).toContain(btn.name);      
    })
  })
  expect(asFragment()).toMatchSnapshot();
});


