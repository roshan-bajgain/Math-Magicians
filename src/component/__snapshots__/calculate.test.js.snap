// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <h2
      class="header"
    >
      Lets Do Some Math! 
    </h2>
    <div
      class="container"
    >
      <div
        class="screen"
      >
        0
      </div>
      <div
        class="buttons"
      >
        <button
          name="Ac"
          type="button"
        >
          Ac
        </button>
        <button
          name="+/-"
          type="button"
        >
          +/-
        </button>
        <button
          name="%"
          type="button"
        >
          %
        </button>
        <button
          class="orange"
          name="÷"
          type="button"
        >
          ÷
        </button>
        <button
          name="7"
          type="button"
        >
          7
        </button>
        <button
          name="8"
          type="button"
        >
          8
        </button>
        <button
          name="9"
          type="button"
        >
          9
        </button>
        <button
          class="orange"
          name="×"
          type="button"
        >
          ×
        </button>
        <button
          name="4"
          type="button"
        >
          4
        </button>
        <button
          name="5"
          type="button"
        >
          5
        </button>
        <button
          name="6"
          type="button"
        >
          6
        </button>
        <button
          class="orange"
          name="−"
          type="button"
        >
          −
        </button>
        <button
          name="1"
          type="button"
        >
          1
        </button>
        <button
          name="2"
          type="button"
        >
          2
        </button>
        <button
          name="3"
          type="button"
        >
          3
        </button>
        <button
          class="orange"
          name="+"
          type="button"
        >
          +
        </button>
        <button
          class="double"
          name="0"
          type="button"
        >
          0
        </button>
        <button
          name="."
          type="button"
        >
          .
        </button>
        <button
          class="orange"
          name="="
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
